### PR TITLE
Add CBW auth class and server group scripts

### PR DESCRIPTION
**Pull request by Herklos**
- Ajout des classes de base au sein d'un module installable via pip
- Ajout de 2 examples correspondants au issue support #102 et #51

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+\.idea/
+__pycache__/
+*.pyc
```
Comment (fwininger) [inline .gitignore]: add a new line

File: README.md [hunk 2]
```diff
@@ -1,2 +1,32 @@
-# cyberwatch_api_toolbox
+# Cyberwatch API toolbox
 Some script to use Cyberwatch Api - www.cyberwatch.fr
+
```
Comment (fwininger) [inline README.md]: Remove this line (double space)

File: README.md [hunk 3]
```diff
@@ -1,2 +1,32 @@
-# cyberwatch_api_toolbox
+# Cyberwatch API toolbox
 Some script to use Cyberwatch Api - www.cyberwatch.fr
```
Comment (fwininger) [inline README.md]: Remove this line

File: cbw_api_toolbox/cbw_api.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+import requests
+from requests import Response
+
+from cbw_api_toolbox.__routes__ import ROUTE_GET_SERVERS
+from cbw_api_toolbox.cbw_auth import CBWAuth
+from cbw_api_toolbox.cbw_objects.cbw_server import CBWServer
+from cbw_api_toolbox.cbw_parser import CBWParser
+
+
```
Comment (fwininger) [inline cbw_api_toolbox/cbw_api.py]: remove this line (double space)

File: cbw_api_toolbox/cbw_auth.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
+import base64
+import datetime
+import hmac
+from hashlib import sha256, md5
+
+from requests.auth import AuthBase
+
+from cbw_api_toolbox import JSON_CONTENT_TYPE, SIGNATURE_HTTP_HEADER, TIMESTAMP_HTTP_HEADER, CONTENT_HASH_HEADER, \
+    CONTENT_TYPE_HEADER, SIGNATURE_HEADER
+
```
Comment (fwininger) [inline cbw_api_toolbox/cbw_auth.py]: Remove this line (double space)

File: cbw_api_toolbox/cbw_parser.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import logging
+from json import JSONDecodeError
+from typing import Dict
+
+
```
Comment (fwininger) [inline cbw_api_toolbox/cbw_parser.py]: double space

File: examples/servers_applications_export.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+        row = 1
+        col = 0
+
+        for application in server.applications:
+            worksheet.write(row, col, application.product)
+            worksheet.write(row, col + 1, application.version)
+            row += 1
+
+exported.close()
```
Comment (fwininger) [inline examples/servers_applications_export.py]: add new line

File: setup.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
+from setuptools import setup
+
+setup(
+    name='cbw-api-toolbox',
+    description='CyberWatch Api Tools.',
+    long_description=open('README.md').read().strip(),
+    version='0.0.1',
+    author='CyberWatch SAS',
+    author_email='contact@cyberwatch.fr',
```
Comment (fwininger) [inline setup.py]: support-it+api@cyberwatch.fr

File: cbw_api_toolbox/cbw_api.py (hunk 4 above)
Comment (Herklos) [reply cbw_api_toolbox/cbw_api.py]: voir https://www.python.org/dev/peps/pep-0008/#blank-lines

File: cbw_api_toolbox/cbw_auth.py (hunk 5 above)
Comment (Herklos) [reply cbw_api_toolbox/cbw_auth.py]: voir https://www.python.org/dev/peps/pep-0008/#blank-lines

File: cbw_api_toolbox/cbw_parser.py (hunk 6 above)
Comment (Herklos) [reply cbw_api_toolbox/cbw_parser.py]: voir https://www.python.org/dev/peps/pep-0008/#blank-lines

Comment (Herklos) [reply cbw_api_toolbox/cbw_parser.py]: :hankey:

File: README.md [hunk 9]
```diff
@@ -1,2 +1,46 @@
...
+Usage
+------------
+
+- To launch a script example :
+    - Install the package (see above)
+    - And then copy the example on your computer from the [examples directory](examples)
+
+- To use this example you will need to edit the script as follow : 
+    - Enter your cyberwatch instance url in ```YOUR_API_URL```
+    <br>*Example : YOUR_API_URL = 'https://edge.cyberwatch.fr'*
```
Comment (fakenine) [inline README.md]: Shouldn't we replace this by a fake URL?

File: README.md (hunk 3 above)
Comment (fwininger) [reply README.md]: Remove `Some script to use Cyberwatch Api - www.cyberwatch.fr` and add `# Cyberwatch Api Toolbox`

File: cbw_api_toolbox/cbw_api.py [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        return f"/{'/'.join(params)}" if params else ""
+
+    def _request(self, route: str, params: List[str] = None):
+        raw_response = None
+        try:
+            raw_response: Response = requests.get(f"{self.api_url}{route}{self._get_params(params)}",
+                                                  auth=CBWAuth(self.api_key, self.secret_key),
+                                                  verify=self.verify_ssl)
+            return json.loads(raw_response.text)
+        except JSONDecodeError as e:
```
Comment (fakenine) [inline cbw_api_toolbox/cbw_api.py]: Shouldn't we catch other common exceptions like timeouts, proxy errors, invalid URLs... ? http://docs.python-requests.org/en/master/_modules/requests/exceptions/

**Comment by hezanathos** [comment]
Coté documentation les changements me conviennent. 👍

File: cbw_api_toolbox/cbw_parser.py (hunk 6 above)
Comment (fwininger) [reply cbw_api_toolbox/cbw_parser.py]: Keep your waste :wastebasket:

**Pull request merged**